Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Update/Internal/FbUpdateSqlGenerator.cs

```
@@ -0,0 +1,364 @@
+            commandStringBuilder.AppendLine("END;");
+            return ResultSetMapping.NotLastInResultSet;
+        } 
+
+        private void AppendUpdateOutputClause(StringBuilder commandStringBuilder)
+        {
+            //Increment of updates 
+            commandStringBuilder
+                    .AppendLine("IF (ROW_COUNT > 0) THEN")
+                    .AppendLine("   AffectedRows=AffectedRows+ROW_COUNT;");
+
+        } 
+
+        private void AppendInsertOutputClause(
+            StringBuilder commandStringBuilder,
+            string name,
+            string schema,
+            IReadOnlyList<ColumnModification> operations,
+            IReadOnlyList<ColumnModification> allOperations)
+        {
+            if (allOperations.Count > 0 && allOperations[0] == operations[0])
+            {
+                commandStringBuilder
+                    .AppendLine($" RETURNING {SqlGenerationHelper.DelimitIdentifier(operations.First().ColumnName)} INTO :AffectedRows;")
+                    .AppendLine("IF (ROW_COUNT > 0) THEN")
+                    .AppendLine("   SUSPEND;");
+            }
+        }
+
+        protected override ResultSetMapping AppendSelectAffectedCountCommand(StringBuilder commandStringBuilder, string name,
+            string schema, int commandPosition)
+        {
+            // Not Implemented!
+            return ResultSetMapping.LastInResultSet;
+        }
+
+        public override void AppendBatchHeader(StringBuilder commandStringBuilder)
+        {
+            //Insert FbSQL Fast(Insert/Update) 
+        }
+
+        protected override void AppendIdentityWhereCondition(StringBuilder commandStringBuilder, ColumnModification columnModification)
+        {
+            // Not Implemented!
+        }
+
+        protected override void AppendRowsAffectedWhereCondition(StringBuilder commandStringBuilder, int expectedRowsAffected)
+        {
+            // Not Implemented!
```

Why not to throw an exception?

---

Yes we can.
Or simply remove.

I've co-opted to leave the scope ready, to make future implementation easy.

In this case there will never be an exetion because we are not using this function yet